
//    Flexisip, a flexible SIP proxy server with media capabilities.
//    Copyright (C) 2010-2015  Belledonne Communications SARL, All rights reserved.
//
//    This program is free software: you can redistribute it and/or modify
//    it under the terms of the GNU Affero General Public License as
//    published by the Free Software Foundation, either version 3 of the
//    License, or (at your option) any later version.
//
//    This program is distributed in the hope that it will be useful,
//    but WITHOUT ANY WARRANTY; without even the implied warranty of
//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//    GNU Affero General Public License for more details.
//
//    You should have received a copy of the GNU Affero General Public License
//    along with this program.  If not, see <http://www.gnu.org/licenses/>.


//option optimize_for = LITE_RUNTIME;

message RecordContactListPb {
  repeated RecordContactPb contact = 1;
}


message RecordContactPb {
  required string uri = 1;
  optional string line_value_copy = 2;
  required uint64 expires_at = 3;
  optional float q = 4 [default = 0];
  required uint64 update_time = 5;
  required string call_id = 6;
  required uint32 cseq = 7;
  required string contact_id = 8;
  optional string route = 9;
  repeated string path = 10;
  repeated string accept_header = 11;
  required bool used_as_route = 12;
}
